src/org/pentaho/reporting/platform/plugin/cache/FileSystemCacheBackend.java

```diff
@@ -42,6 +45,7 @@
   private static final Log logger = LogFactory.getLog( FileSystemCacheBackend.class );
   public static final String REPLACEMENT = "_";
   public static final String SLASHES = "[/\\\\]+";
+  private final Map<List<String>, Object> syncMap = new HashMap<>();
```

Comment:
Access to that map is not thread-safe. You synchronize once you have the lock, but the work of looking up the lock is not threadsafe. Use concurrent-hashmap here? Then you can write the updates in a way that is thread-safe in your "getLock(..)" method. A "putIfAbsent" call will make that operation safe and fast.